### DISCUSSION TITLE
Support Multitenancy

### DISCUSSION THREAD

**Discussion by maukoese** [spatie/laravel-settings#134] [Ideas]
I'm using the Tenancy for Laravel package.

Would be nice to get this package to run the settings migrator to the new database created.

Thank you so much.

**Comment by jhagan-91** [comment]
This package wouldn't prevent that.

**Comment by maukoese** [comment]
I created an event listener to run the migrations in the tenant database.

**Reply by tomasvanrijsse** [reply]
I just moved the SettingsMigration to `/database/migrations/tenant/`.
When I kept them in `/database/settings` the migrations where being fired when I run `artisan migrate` for the 'central' database.
So I needed to prevent running them on the central database and only run them in the tenant database.
Don't you have that problem with your setup @maukoese ?

I am wondering though: why did Spatie make a separate directory for these migration instead of just storing them in the default migrations folder?

**Reply by rubenvanassche** [reply]
A separate directory was created because these settings migrations are completely different from database migrations, we wanted to keep these apart. But you can always change this behaviour by changing the config `migrations_paths` key. Not sure how this all works with Tenancy.
